_docs/automation-time-apis.md

```diff
@@ -121,6 +121,180 @@ pub enum Error {
+    executions: u32,
+    )
+```
+
+** The 4 enums for AutomationAction are: Notify, NativeTransfer, XCMP, AutoCompoundDelegatedStake.
+** Note that this RPC does not return the inclusion fee of including the task onto the block. That is a separate RPC call: payment.queryFeeDetails. This only provides the execution fee of the task. 
+** The difference between inclusion fee and execution fee is created in the fact that the OAK blockchain runs tasks in the future off of a time trigger. This means that fees are charged not only for inclusion of the task onto the chain, but also the execution of the task in the future. This current RPC only shows the execution cost.
+
+#### Errors
+```rust
+pub enum Error {
+    // The fee calculation went wrong
+    "Unable to get time automation fees",
+    // The fee amount is too large and cannot fit in u64 representation.
+    "RPC value doesn't fit in u64 representation",
+}
+```
+
+#### Example
+Sample Request
+```cURL
+curl --location --request POST 'http://rpc.turing-staging.oak.tech' \
+--header 'Content-Type: application/json' \
+--data-raw '{"id":1, "jsonrpc":"2.0", "method": "automationTime_getTimeAutomationFees", "params": ["Notify", 3]}'
+```
+
+Sample Response
+```JSON
+{
+    "jsonrpc": "2.0",
+    "result": 252000000,
+    "id": 1
+}
+```
+
+### RPC: Calculate Optimal Autostaking
+This API calculates the optimal parameters for autostaking. Autostaking is the process of restaking the returns from some staked principal. Restaking too frequently becomes too costly for fees, but restaking too infrequently leaves too much opportunity cost on the table, as the unstaked balance sits in the wallet without providing additional staking value. This function finds the optimal frequency and the corresponding APY for restaking.
+
+#### Call
+```rust
+fn calculate_optimal_autostaking(
+    /// The amount that will be staked.
+    principal: i128,
+    /// The collator to which the principal will be staked.
+    collator: AccountId,
+    )
+```
+
+** This RPC will return a JSON object with a time period and an APY. The time period represents the optimal staking duration and the APY is the projected staking return.
+
+#### Errors
+```rust
+pub enum Error {
+    // The optimal autostaking calculation went wrong.
+    "Unable to calculate optimal autostaking",
+    // The collator provided is not a real collator.
+    "collator does not exist",
+    // The fee calculation went wrong.
+    "could not calculate fee",
+}
+```
+
+#### Example
+Sample Request
+```cURL
+curl --location --request POST 'http://rpc.turing-staging.oak.tech' \
+--header 'Content-Type: application/json' \
+--data-raw '{"id":1, "jsonrpc":"2.0", "method": "automationTime_calculateOptimalAutostaking", "params": [1000000000000, "6AwtFW6sYcQ8RcuAJeXdDKuFtUVXj4xW57ghjYQ5xyciT1yd"]}'
+```
+
+Sample Response
+```JSON
+{
+    "jsonrpc": "2.0",
+    "result": {
+        "period": 63,
+        "apy": 0.08711145708327778
```

Comment:
Same here. Is this `8% apy` or `.08` apy? @atomaka

Reply:
8%!